libvast/test/type.cpp

```diff
@@ -721,6 +721,25 @@ TEST(enriched types) {
   CHECK_EQUAL(lat, at);
 }
 
+TEST(aliases) {
+  const auto t1 = bool_type{};
+  const auto t2 = type{"quux", t1};
+  const auto t3 = type{"qux", t2, {{"first"}}};
+  const auto t4 = type{"baz", t3};
+  const auto t5 = type{t4, {{"second"}}};
+  const auto t6 = type{"bar", t5, {{"third"}}};
+  const auto t7 = type{"foo", t6, {{"fourth"}}};
+  auto aliases = std::vector<type>{};
+  for (auto&& alias : t7.aliases())
+    aliases.push_back(std::move(alias));
+  REQUIRE_EQUAL(aliases.size(), 5u);
+  CHECK_EQUAL(aliases[0], t6);
+  CHECK_EQUAL(aliases[1], t4);
```

Comment:
Is `t5` skipped on purpose?

Reply:
Yes. The constructor of `type` folds it into the next named layer on construction (see #2138). This is important because it is not something a user can actually express in configuration.

Reply:
Okay, not easy to see from this context, but fine with me.